discovery-infra/tests/base_test.py

```diff
@@ -25,13 +22,27 @@ def random_name():
 class BaseTest:
 
     @pytest.fixture(scope="function")
-    def nodes(self, setup_node_controller):
-        controller = setup_node_controller
-        nodes = Nodes(controller, env_variables["private_ssh_key_path"])
-        nodes.set_correct_boot_order(start_nodes=False)
-        yield nodes
-        nodes.shutdown_all()
-        nodes.format_all_disks()
+    def nodes(self, setup_node_controller, is_dev_env):
+        if is_dev_env:
+            asset = None
+            try:
+                asset = Assets()
```

Comment:
i think this way better cause it is much less code and not try finally in IF scope

```
@pytest.fixture(scope="function")
def nodes(self, setup_node_controller):
    asset = None
    try:
        if is_dev_env:
            asset = assets.Assets()
            env_variables["asset"] = asset.get_asset()
        controller = setup_node_controller(**env_variables)
        controller.prepare_nodes()
        nodes = Nodes(controller, env_variables["private_ssh_key_path"])
        yield nodes
        logging.info(f'--- TEARDOWN --- node controller\n')
        controller.destroy_all_nodes()
    finally:
        if asset:
            asset.return_all()
```